Mc core style import swap

micahchiang opened the pull request:
This PR swaps the core style import from formation to css-library.

Sibling css-library PR - https://github.com/department-of-veterans-affairs/component-library/pull/1359

**Note on Injected Header style changes**
The swap from formation to css-library introduces a change to how utility classes are given values. Previously in Formation, utility classes used a base multiple of 8px. In css-library, utility classes rely on `rem` values. This matters because Teamsites **still use 10px as a root font size.**. What this means is that utility classes needed to be updated for various element class lists. This can be seen in the changes files where for example some elements went from something like `vads-u-padding--2` to `vads-u-padding--3`. The implication here is that computed values for the injected header will be changing by a minuscule value: in most instances, 0.5px. In some cases, 1px. 

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

Did you change site-wide styles, platform utilities or other infrastructure?
- [ ] No
- [x] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
In an effort to sunset [Formation](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/tree/master/packages/formation), this pull request updates the import path for VA.gov stylesheets to [css-library](https://github.com/department-of-veterans-affairs/component-library/tree/main/packages/css-library). 

Several files have been added to vets-website for features that will be migrated once this PR is merged. In particular, grid classes and the sitewide footer. 

## Related issue(s)

- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3397

## Testing done
Visual QA was conducted between a locally running instance of this work against staging and production. Screenshots were captured and added below for high trafficked parts of VA.gov as well as applications that span the breadth of services offered on VA.gov. Screenshots include mobile and desktop diffs. 

Unit tests and cypress e2e tests were also used to verify functionality didn't regress, and any visual regressions that were flagged by CI runs were addressed, and the build is now passing successfully. 

## Screenshots

<details><summary>Profile</summary>

Production:
![image](https://github.com/user-attachments/assets/ece7599d-de7d-4ec3-81af-4ce94fa5aaf3)

Local:
![image](https://github.com/user-attachments/assets/009fc5bf-3077-491c-a1ba-f30daa19c831)

Production:
![image](https://github.com/user-attachments/assets/b5129610-ff81-4141-b327-c76f66d5056b)

Local:   
![image](https://github.com/user-attachments/assets/d7bb9183-4135-4d2c-b8b0-54ca180345c7)

Production:
![image](https://github.com/user-attachments/assets/b6b1905d-1460-4683-8f6a-08abd83faf15)

Local:
![image](https://github.com/user-attachments/assets/6b172f84-c7d3-4569-b4b1-a139b26d585b)

Production:
![image](https://github.com/user-attachments/assets/62ea5a56-bcc3-4e99-9428-aacd51c20c5d)

Local:
![image](https://github.com/user-attachments/assets/a4e6402b-c5f8-412d-96ea-61090e5138e9)

Production:
![image](https://github.com/user-attachments/assets/22f88634-bd07-48e2-955d-13d13a4aa73e)

Local:
![image](https://github.com/user-attachments/assets/b19ba312-02c3-473a-938f-8a1f2a15c714)

Production:
![image](https://github.com/user-attachments/assets/e9cc26c8-2a80-4e6d-a84f-ede96a4d1ce0)

Local:
![image](https://github.com/user-attachments/assets/86a1033b-e75d-40fb-868c-8ca0154cc567)

Production:
![image](https://github.com/user-attachments/assets/c70f1772-0ce5-4260-8de4-b219765865f3)

Local:
![image](https://github.com/user-attachments/assets/e0a45440-704d-4970-81ed-cd9bbbba4974)

Production:
![image](https://github.com/user-attachments/assets/804a589f-70da-4a3b-8b77-7ae68c8f9bad)

Local:
![image](https://github.com/user-attachments/assets/302b5472-ad30-4218-adab-7e4145db8e47)


</details>

<details><summary>MHV, local on left, staging on right</summary>

![image](https://github.com/user-attachments/assets/ee3a560b-72a6-45a0-9d57-78bfcbbc6899)

![image](https://github.com/user-attachments/assets/eb30a060-bb77-4642-b153-d72fc1b994ac)

![image](https://github.com/user-attachments/assets/5e81edd4-6ef8-4870-93cb-08fcdc9a9bf9)

![image](https://github.com/user-attachments/assets/52f9f878-039d-4a5d-9306-c6ca982e51e3)

![image](https://github.com/user-attachments/assets/a0adef5f-0c88-4c28-a847-b72f1ad1e47f)

![image](https://github.com/user-attachments/assets/5fcf4e72-38bf-4a4d-910d-290587193bcf)

![image](https://github.com/user-attachments/assets/c67e31f5-cd44-4f7d-b368-a2217a378b06)

![image](https://github.com/user-attachments/assets/601c8902-679c-4f29-9929-3903b0e71053)

![image](https://github.com/user-attachments/assets/e765fe94-532a-4298-ac9d-8f15abeb4feb)

![image](https://github.com/user-attachments/assets/9da44f31-e22e-46c1-9959-e1c70c0ab7ab)

![image](https://github.com/user-attachments/assets/9b6fadf9-6c4f-42b7-92c0-0bf4bd7e289f)

![image](https://github.com/user-attachments/assets/135bacbd-deb3-486f-88fa-bbd0dcd3ae5d)

![image](https://github.com/user-attachments/assets/2b18e802-4fef-4105-8570-2eaa77651bd2)

![image](https://github.com/user-attachments/assets/feaec275-70e7-40cd-901d-ef5565e7abe0)

![image](https://github.com/user-attachments/assets/c3a7c8b5-7ac5-477e-bc95-c3a4a1bada16)

![image](https://github.com/user-attachments/assets/f3cd1378-1de3-40ac-bc4f-eb67b0d545e7)

![image](https://github.com/user-attachments/assets/5901289e-1742-44e0-b3e9-7633201b54d2)

![image](https://github.com/user-attachments/assets/3d5c68e9-99e0-4b10-b0f1-8c0a1ee8563a)

</details>

<details><summary>Combined Debt Portal, local on left staging on right</summary>

![image](https://github.com/user-attachments/assets/52edbbb4-3a0a-4a63-b330-a898753ec737)

![image](https://github.com/user-attachments/assets/5cb0244c-dfd9-48a3-bc17-416fce670853)

![image](https://github.com/user-attachments/assets/9dc24474-e3ab-46bc-a564-b2d389d1ed22)

![image](https://github.com/user-attachments/assets/e92bf82b-8ebe-4c5b-9ff9-8e531a4edbde)

</details>

<details><summary>Claims Status, local on left, staging on right</summary>

![image](https://github.com/user-attachments/assets/6bea79a2-ceda-4c65-a141-7158ba916ae1)

![image](https://github.com/user-attachments/assets/8cf3d598-e337-42cf-bf15-3b637ba31ded)

![image](https://github.com/user-attachments/assets/b5c11b2a-8c1a-426c-ab6f-ca0e51603ecd)

![image](https://github.com/user-attachments/assets/c32c1163-26cc-4ff8-a0b5-e9bf8ffc84b3)

![image](https://github.com/user-attachments/assets/f6bf2c0b-cb93-4ef6-acbe-952d958b9670)

![image](https://github.com/user-attachments/assets/739b3564-eb8a-48f9-b68c-4748210bad00)

![image](https://github.com/user-attachments/assets/e5835dd0-efdc-4af9-bf1e-0a3fb9216bdd)

![image](https://github.com/user-attachments/assets/f6746342-b8c4-4b50-ace6-4a66935b6c5d)

![image](https://github.com/user-attachments/assets/d60256c9-4f42-4f71-bbf2-f2f9d4038d0d)

![image](https://github.com/user-attachments/assets/92c82191-3ddc-4aad-9b5d-486409ac94b5)

![image](https://github.com/user-attachments/assets/d411a11e-6c71-490f-96bc-19b8fa0b2fee)

![image](https://github.com/user-attachments/assets/32cbe03c-2ce1-47b5-8a2e-a9cc0f03ccdb)

![image](https://github.com/user-attachments/assets/a904e5f3-463b-4484-8209-a145b88c7329)

![image](https://github.com/user-attachments/assets/cfdcd557-97ad-46f1-9375-3a711aeee2bc)

![image](https://github.com/user-attachments/assets/07e4f6ec-7845-4336-99d6-582305d44bbf)

![image](https://github.com/user-attachments/assets/bb3c87b8-e612-424f-91ba-32ff287ec3a1)


</details>

<details><summary>Injected header, local left, production right</summary>

![Screenshot from 2024-11-08 13-47-57](https://github.com/user-attachments/assets/6ade6f52-5e24-4ff5-b41c-e97d9461e9be)
![Screenshot from 2024-11-08 13-47-42](https://github.com/user-attachments/assets/9a8a9ae1-b67f-44bf-97f4-e17810db36ae)
![Screenshot from 2024-11-08 13-47-13](https://github.com/user-attachments/assets/5df63012-7184-44f4-973b-deaecea7f068)
![image](https://github.com/user-attachments/assets/193cc72c-9c92-4c71-ba83-3a49a30e73d3)
![image](https://github.com/user-attachments/assets/257e20a7-ab59-4927-9dd2-ca056ec71ab9)
![image](https://github.com/user-attachments/assets/25b9a8c1-fedc-48b6-8ca8-275a50534c31)
![image](https://github.com/user-attachments/assets/41fd30af-9e4a-454e-bea4-8727e52164a5)
![image](https://github.com/user-attachments/assets/42318472-93a1-44aa-ae51-3848fdb0aaee)
![image](https://github.com/user-attachments/assets/337c2ccf-2718-47a4-8e32-baabcc33fbaa)
![image](https://github.com/user-attachments/assets/184ea8e8-4246-47dc-a24e-91e506ae090f)
![image](https://github.com/user-attachments/assets/331ed20c-1140-440d-bf7e-e9ffc07ad7da)
![image](https://github.com/user-attachments/assets/639ce67c-a088-46e2-a03e-00aed63510d5)
![image](https://github.com/user-attachments/assets/b967737d-178e-4eee-b5fa-6344e696cc02)
![image](https://github.com/user-attachments/assets/759f98c5-e5e1-432c-9e84-f4c72b1d4ecb)
![image](https://github.com/user-attachments/assets/ddd12638-5a69-4055-a5d3-8e5b2a7913bb)
![image](https://github.com/user-attachments/assets/3ceff315-b15d-43c9-9289-157df50237a8)


</details>

<details><summary>Mock Form Flow, local left, staging right</summary>

![image](https://github.com/user-attachments/assets/411e139b-9941-499f-8cb4-2b9961857c05)
![image](https://github.com/user-attachments/assets/2f67876c-e99a-4ef1-b8cd-c64e1f909318)
![image](https://github.com/user-attachments/assets/0c9a1c5f-2cb2-409c-8f10-dce52fa28bb5)
![image](https://github.com/user-attachments/assets/e4c48b43-33a0-4f4b-9c95-f25357b5e3a3)
![image](https://github.com/user-attachments/assets/07d5466e-8912-44df-a0d7-c2606a888704)
![image](https://github.com/user-attachments/assets/4a6cefb5-21bd-4dc6-8926-43fa13c91d66)
![image](https://github.com/user-attachments/assets/f0fb8aca-f799-4ce9-9aa1-c329f8ad33bf)
![image](https://github.com/user-attachments/assets/374c3841-1139-4543-88dc-40c5a12b2f8a)
![image](https://github.com/user-attachments/assets/09204637-0659-42f2-a194-5a90b46c0335)
![image](https://github.com/user-attachments/assets/882b536f-a732-4415-aa15-300ef6537171)
![image](https://github.com/user-attachments/assets/0cd866da-3ae4-424b-a22f-bc5b7e0712b2)
![image](https://github.com/user-attachments/assets/351b761b-21a9-4ff8-b458-8650db2e0400)
![image](https://github.com/user-attachments/assets/b22e9729-a3f4-44b8-a3b1-801160ec307b)
![image](https://github.com/user-attachments/assets/65794f6b-c36f-43fd-8dc5-4a08d301c94e)
![image](https://github.com/user-attachments/assets/1615e567-16f7-425e-8122-e134c1b1413d)
![image](https://github.com/user-attachments/assets/12c4d1e3-b0e2-44fe-b144-bf0600e4c02a)
![image](https://github.com/user-attachments/assets/e1058228-a132-4167-bfd3-c3e676576216)
![image](https://github.com/user-attachments/assets/15bcfe3f-e27f-4bb5-adae-77d123d67547)
![image](https://github.com/user-attachments/assets/1b673cac-656d-4dd0-bbf0-d559dc3c8b75)
![image](https://github.com/user-attachments/assets/dc6875f6-6ef6-4a95-944f-aed1c91d6597)
![image](https://github.com/user-attachments/assets/a718002c-5c18-424c-9cf2-0818762b7c0d)
![image](https://github.com/user-attachments/assets/98c77161-1ed2-47bc-828b-12dc9a527136)
![image](https://github.com/user-attachments/assets/45eb5267-94a6-4144-8e8b-a12671eb5f1b)
![image](https://github.com/user-attachments/assets/dee25560-8551-4cc4-975e-3bef3171a5e3)
![image](https://github.com/user-attachments/assets/067eedec-4297-4a9d-bae6-37f631fb76a2)
![image](https://github.com/user-attachments/assets/512687ca-6135-40dc-a12a-e7cc4f03c2ab)
![image](https://github.com/user-attachments/assets/bdf26234-457d-44f9-a2d4-ffdb2f204029)
![image](https://github.com/user-attachments/assets/18994a60-61b0-4220-a558-9de27ecc87ca)
![image](https://github.com/user-attachments/assets/83734429-fac9-4a1c-ab27-8002d412dcce)
![image](https://github.com/user-attachments/assets/3788a261-c6d3-4796-8c0e-acebdee184c9)
![image](https://github.com/user-attachments/assets/a70809af-018e-41e8-ad56-1e4bd2a0954f)
![image](https://github.com/user-attachments/assets/40828525-f9b5-4780-9d37-3eeafb27ebd4)
![image](https://github.com/user-attachments/assets/465b77b0-bf27-4966-8455-4ec0c9700d5b)
![image](https://github.com/user-attachments/assets/80e3ce96-5f44-431a-9454-7f60bdee2672)
![image](https://github.com/user-attachments/assets/385ecba5-35ca-47ba-97be-2d411b897b59)


</details>

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

micahchiang committed:
initial

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
update logo image

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
re-add typography stylesheet

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
update

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
update order

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
put grid in shame for now

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
update imports

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
remove extra shame import

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
package.json for now

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
add img default styles to shame

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang commented on package.json:
```diff
@@ -306,6 +306,7 @@
     "octokit": "^2.1.0",
     "octokit-plugin-create-pull-request": "^3.11.0",
     "prop-types": "^15.6.1",
+    "querystring-es3": "^0.2.1",
```
needed because we're still relying on uswds v1 grid styles while we get off of Formation the repo.

micahchiang commented on src/platform/site-wide/sass/shame.scss:
```diff
@@ -636,3 +636,92 @@ div[data-widget-type="events"] {
 }
 /* ^^^ DO NOT REMOVE w/o DS REVIEW: solves !important conflict*/
+
+/* ~~~~~ DO NOT REMOVE w/o DS REVIEW: Temporary fixes to enable core.scss swap ~~~~~~ */
```
For the sake of keeping the mental model relatively simple, I'm just moving our grid classes into shame for the time being. I know I know, we want to get rid of shame.scss eventually. This makes testing easier. These grid classes will 100% be moved to css-library when we have the capacity to update all of grid.

micahchiang commented on src/platform/site-wide/sass/shame.scss:
```diff
@@ -636,3 +636,92 @@ div[data-widget-type="events"] {
+
+// foundation/global.scss sets display and vertical align on img tags. This is most obviously seen in the footer image
+// in content-build
+img {
+  display: inline-block;
+  vertical-align: middle;
```
Foundation sets these on the element selector. Without these styles, the image in `.va-banner-container` right above the footer won't sit flush to the rest of the footer. Also without these, images in promo blocks will create some minor content shift in elements above them.

micahchiang committed:
package.json

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
Merge branch 'main' into mc-core-style-test

powellkerry committed:
add missing foundation-site img styles that constrained the width of images to 100%

powellkerry commented:
Team sites, after changes, look normal:

![Screenshot 2024-10-21 at 8 40 51 AM](https://github.com/user-attachments/assets/8f7c0fdf-faaf-410f-9845-766efa77263f)

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
remvoe shared import

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
add v1-grid

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

powellkerry committed:
fix media queries for v1 grid

micahchiang committed:
hardcode rem values in grid for now

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

powellkerry committed:
add missing import for scale-rem function

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
update css-library version

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

powellkerry committed:
install querystring

powellkerry committed:
fix failing cypress tests

powellkerry committed:
fix cypress test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

powellkerry committed:
Merge branch 'main' into mc-core-style-test

powellkerry committed:
Update css-library version for primary-button styles

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
updates to v1-grid to fix breakpoint columns

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
update package.json

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
update injected header styles

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang commented on src/applications/proxy-rewrite/sass/style-consolidated.scss:
```diff
@@ -226,6 +226,12 @@
+    // overriding utility class because teamsites has a root font-size of 10px and we don't have a utility class 
+    // in rem that computes exactly to that. 
+    .medium-screen\:vads-u-padding--2 {
+      padding: 16px !important;
```
needed because teamsites still has a root font-size of 10px

micahchiang commented on src/applications/proxy-rewrite/partials/mobile/header/index.jsx:
```diff
@@ -127,7 +127,7 @@ const MobileHeader = ({ isDesktop, megaMenuData }) => {
 
       <nav className="vads-u-display--flex vads-u-flex-direction--column vads-u-margin--0 vads-u-padding--0">
-        <div className="header-logo-row vads-u-background-color--primary-darker vads-u-display--flex vads-u-align-items--center vads-u-justify-content--space-between vads-u-padding-y--1p5 vads-u-padding-left--1p5 vads-u-padding-right--1">
+        <div className="header-logo-row vads-u-background-color--primary-darker vads-u-display--flex vads-u-align-items--center vads-u-justify-content--space-between vads-u-padding-y--2p5 vads-u-padding-left--2p5 vads-u-padding-right--1p5">
```
This and the below adjustment are needed because teamsites still has a root font-size of 10px and updating these classes gets us as close as we can get with css-library utilities.

micahchiang committed:
fix gov banner display on mobile

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang commented on src/applications/proxy-rewrite/partials/mobile/header/index.jsx:
It seems like some of the display logic in the mobile injected header was lost when formation core was removed, which led to the expanded gov banner always being displayed. The changes in this file add back in the toggle functionality and update the aria-hidden attribute as needed.

micahchiang committed:
update injected header styles

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
resolve main

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
add foundation grid to v1-grid stylesheet

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

powellkerry committed:
Add missing icons, re-order v1-grid import

micahchiang committed:
remove whitespace

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
update import order

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
fix margins on find va forms

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

powellkerry committed:
Update css-library version for max-width fix

powellkerry committed:
revert e2e test fix after max-width adjustment

powellkerry committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

powellkerry committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
remove minimal.scss

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
Merge branch 'mc-core-style-test' of github.com:department-of-veterans-affairs/vets-website into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang commented on package.json:
```diff
@@ -263,7 +263,7 @@
     "@department-of-veterans-affairs/formation": "^12.0.0",
     "@department-of-veterans-affairs/component-library": "^48.0.2",
-    "@department-of-veterans-affairs/css-library": "^0.15.0",
+    "@department-of-veterans-affairs/css-library": "^0.14.0-rc7",
```
Once [this PR](https://github.com/department-of-veterans-affairs/component-library/pull/1359) has been merged in component-library, the dependency here will be updated with a non-release candidate version.

micahchiang commented on package.json:
```diff
@@ -305,6 +305,7 @@
     "octokit": "^2.1.0",
     "octokit-plugin-create-pull-request": "^3.11.0",
     "prop-types": "^15.6.1",
+    "querystring": "^0.2.1",
```
For now, we're still relying on some grid styles from uswds v1 and foundation-sites, which have a dependency on querystring. When we migrate grid to uswds v3, this dependency will no longer be needed.

micahchiang commented on src/applications/proxy-rewrite/partials/mobile/header/mega-menu/index.jsx:
Changes in here are for the same reason mentioned above: teamsites still relies on a root font-size of 10px, so all utility classes needed to be incremented to get as close as possible to what's in production now. "As close as possible" here mostly means within 0.5px, occasionally 1px.

micahchiang committed:
remove formation font and image paths

Signed-off-by: Micah Chiang <micahkchiang@gmail.com>

micahchiang committed:
Merge branch 'mc-core-style-test' of github.com:department-of-veterans-affairs/vets-website into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

randimays reviewed (commented):
I ran this locally and looked at `/discharge-upgrade-instructions` and noticed the fonts were not being picked up correctly.

<img width="1020" alt="Screenshot 2024-11-26 at 9 33 27 AM" src="https://github.com/user-attachments/assets/193d4354-d512-456d-bb10-325b07f59f04">

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang commented:
@randimays - Could you tell me how you ran things locally? I see fonts rendering correctly against production after running a yarn install in vets-website, a yarn build in content-build, and then a yarn watch in content-build: 
![image](https://github.com/user-attachments/assets/c53ddc97-5e01-41e9-ad69-0787cdec9917)

randimays commented:
Ah thank you @micahchiang I didn't realize there were package.json updates; thought it was just import swaps. I can see the correct fonts now! 🙏🏻

randimays reviewed (commented):
Reviewing a little bit at a time and found a few issues with Discharge Upgrade Wizard and Facility Locator so far.

### /discharge-upgrade-instructions 

Missing some "table" formatting in the review section and there are some page width differences.

| Local | Prod |
| --- | --- |
| <img width="629" alt="Screenshot 2024-11-26 at 10 52 43 AM" src="https://github.com/user-attachments/assets/58ec96c9-1bf5-49fa-ab38-959e8d8ecd71"> | <img width="746" alt="Screenshot 2024-11-26 at 10 52 35 AM" src="https://github.com/user-attachments/assets/e8e4862e-73a5-477e-8962-8c990d89021d"> |
| <img width="766" alt="Screenshot 2024-11-26 at 10 54 33 AM" src="https://github.com/user-attachments/assets/79e5b25e-3d1b-4428-a6c0-c484ff1a0df8"> | <img width="891" alt="Screenshot 2024-11-26 at 10 54 36 AM" src="https://github.com/user-attachments/assets/eda8cd2e-b752-418f-81eb-69b9257ab2aa"> |

### /find-locations

Missing the up/down arrow in the dropdown for `Service type`.

| Local | Prod |
| --- | --- |
| <img width="1046" alt="Screenshot 2024-11-26 at 11 12 48 AM" src="https://github.com/user-attachments/assets/7615790f-af7d-4f1b-90ae-9744c8a6d9a9"> | <img width="1030" alt="Screenshot 2024-11-26 at 11 12 51 AM" src="https://github.com/user-attachments/assets/67dd3ada-4c89-4bed-85d8-c59e72b6feea"> |

micahchiang commented:
@randimays - I think there's a discrepancy between local, staging, and prod for the discharge wizard. I noticed it while looking at markup structure because classes and elements aren't equal. It seems like production is rendering something different, based on this code in the discharge wizard: 

```javascript
export default function DischargeWizardApp({ children }) {
  const isProd = environment.isProduction();

  if (isProd) {
    return (
      <div className="discharge-wizard row vads-u-padding-x--1 desktop-lg:vads-u-padding-x--0">
        <Breadcrumbs />
        {children}
      </div>
    );
  }
  return (
    <div className="row discharge-wizard-v2 vads-u-padding-bottom--8">
      <BreadcrumbsV2 />
      <div className="usa-width-two-thirds medium-8 columns">{children}</div>
    </div>
  );
}
```

Local looks identical to staging for the discharge flow you mentioned above: 
<details><summary> Local vs Staging Discharge Flow </summary>

![image](https://github.com/user-attachments/assets/baa6c1c4-35a4-44b3-a1d2-444c4ce72262)
![image](https://github.com/user-attachments/assets/d7b04404-fec6-4448-a581-d18e8e9f124f)
</details>

If I update the discharge wizard to always just return the markup from the isProd if statement, then local looks like production:
<details><summary>Local vs Production Discharge Flow </summary>

![image](https://github.com/user-attachments/assets/304d0314-e563-4ec8-bcc2-4890ece2d673)
![image](https://github.com/user-attachments/assets/5df3f336-87b9-4ef4-9904-fafcd0eeafd6)
</details>

micahchiang commented:
@randimays - I'm unable to reproduce the missing arrows from `/find-locations`: I see them locally: 

![image](https://github.com/user-attachments/assets/e8897c92-5d76-4fd2-b947-c00498354620)

randimays commented:
@micahchiang Yes, there is weirdness in the Discharge Upgrade Wizard across environments. We are about to launch the newest DUW and it's currently living in staging (you can't get to the old one AFAIK), and has some toggles to render locally and in review instances. I didn't think it affected layout but I'll review that. I'll see what's going on with my local build; perhaps my branches are in weird states and need to be completely refreshed. Apologies for the goose chase(s)!

randimays reviewed (commented):
Went through some more testing after refreshing the branches / builds. Some observations:

1. The fonts on the footer on mobile are not rendering correctly.

| Local | Prod |
| --- | --- |
| <img width="399" alt="Screenshot 2024-11-27 at 7 00 11 AM" src="https://github.com/user-attachments/assets/7d1e6364-efe5-44de-bb7e-e551d1465c84"> | <img width="400" alt="Screenshot 2024-11-27 at 7 00 18 AM" src="https://github.com/user-attachments/assets/31512193-66ea-42f4-a496-47288adf1143"> |

2. Facility locator `Service type` arrows are still missing. `/find-locations`

| Local | Prod |
| --- | --- |
| <img width="441" alt="Screenshot 2024-11-27 at 7 02 11 AM" src="https://github.com/user-attachments/assets/490714cb-e9a0-4c21-8a16-070613190135"> | <img width="441" alt="Screenshot 2024-11-27 at 7 02 17 AM" src="https://github.com/user-attachments/assets/d330a00f-a6c0-4085-80cf-c396fecf9774"> |

3. The page width on several pages is different. Shown here at `/find-forms`. Also is different at `/resources`

| Local | Prod |
| --- | --- |
| <img width="999" alt="Screenshot 2024-11-27 at 7 09 05 AM" src="https://github.com/user-attachments/assets/26aaa6b9-3a48-4495-a540-8d20625f4856"> | <img width="998" alt="Screenshot 2024-11-27 at 7 08 54 AM" src="https://github.com/user-attachments/assets/c02e4d0a-0866-4557-aaf4-d937b3f44e7d"> |

4. Page width is different on `/search` too, and the underline under `More VA search tools` is missing

| Local | Prod |
| --- | --- |
| <img width="1207" alt="Screenshot 2024-11-27 at 7 16 49 AM" src="https://github.com/user-attachments/assets/7aeea5a9-35fb-4652-bd86-02d72f5a1bcf"> | <img width="1208" alt="Screenshot 2024-11-27 at 7 16 54 AM" src="https://github.com/user-attachments/assets/dd704fcf-35b9-48a6-a708-686e90d31a0f"> |

5. Mobile search button in the header is misaligned

| Local | Prod |
| --- | --- |
| <img width="400" alt="Screenshot 2024-11-27 at 7 18 50 AM" src="https://github.com/user-attachments/assets/faf86e6a-672c-438e-8753-b5a95e4a8a27"> | <img width="399" alt="Screenshot 2024-11-27 at 7 18 55 AM" src="https://github.com/user-attachments/assets/482d78b0-e2d0-4e3e-86e3-b7e01d60cdde"> |

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang commented:
@randimays See comments and updates below:

#### Footer has been fixed, thanks for catching that!
<details><summary>Local vs Prod computed font match</summary>

![image](https://github.com/user-attachments/assets/8293eada-b28b-4ca0-a8a7-d57e893cbc7a)
</details>

#### Select arrows on find-locations
I still see the arrows rendered correctly in the find-locations form. It's strange because the asset for that is definitely there. It's included in the changes files of this PR :thinking: Maybe the review instance will shed some more light on this once its ready.

#### Search button on mobile has been fixed. Thanks for catching that!
<details><summary>Local vs Prod search button icon realigned</summary>

![image](https://github.com/user-attachments/assets/6758c5c2-3c98-4f22-8f0f-aa9af343a3d2)

</details>

#### Width on /find-forms and /resources
We were missing a rule for `usa-content`. I've added it, and now computed max-width matches production.
<details><summary>Local vs Prod max-width </summary>

![image](https://github.com/user-attachments/assets/625d85d3-de0f-4ca6-ad4c-f4933f0274e1)

</details>

#### Search Page width 
Formation had a custom override specifically for `.usa-width-three-fourths` which we missed while moving things. I've added an override for it to our v1-grid file in vets-website. We were also missing style rules for the `.highlight` class. Width is fixed and the blue underline is back.
<details><summary>Local vs Prod Search page</summary>'

![image](https://github.com/user-attachments/assets/50551b8d-b283-4ca3-81be-8a7a195718cc)

</details>

micahchiang committed:
fixes based on sitewide feedback

Signed-off-by: Micah Chiang <15097156+micahchiang@users.noreply.github.com>

randimays commented:
@micahchiang Thank you, those fixes look good! I checked the review instance for Facility Locator and the arrows on that dropdown are missing for me 🤔 

I thought I would check the injected header/footer just in case (I know DST owns it but old habits die hard). Couple of things I noticed below. I think maybe some of them are related (classes/styles missing or needing to be moved around).

1. Mobile - The gray banner and red banner are shorter vertically and the dark blue section with the logo has more padding on the bottom than the top
2. Mobile - When the gray banner is open, there is a spacing difference between the trigger and the text below
3. Tablet/Desktop - the gray banner "Here's how you know" font is a bit different
4. Tablet - the gray banner page margins are more narrow
5. Mobile - The search bar input has a minor spacing issue with the Search text and the input, and the input and button are a bit misaligned.
6. Tablet - space between the blue footer and the bottom gray bar is narrower

| Local | Prod |
| --- | --- |
| <img width="400" alt="Screenshot 2024-11-27 at 2 30 42 PM" src="https://github.com/user-attachments/assets/e8a1b5e0-091f-4c1e-91bb-89cf6ee4d044"> | <img width="400" alt="Screenshot 2024-11-27 at 2 30 47 PM" src="https://github.com/user-attachments/assets/ae814cda-fb67-4a38-9980-97025036ef7f"> |
| <img width="399" alt="Screenshot 2024-11-27 at 2 51 26 PM" src="https://github.com/user-attachments/assets/d21e6f83-5a8a-4b33-bc48-323398ae60ec"> | <img width="399" alt="Screenshot 2024-11-27 at 2 51 32 PM" src="https://github.com/user-attachments/assets/cd6a541a-431f-4d1b-bfec-1fa73c32cd09"> |
| <img width="441" alt="Screenshot 2024-11-27 at 2 55 00 PM" src="https://github.com/user-attachments/assets/a733a82d-f96a-4efc-9ace-9a82abedc6bb"> | <img width="438" alt="Screenshot 2024-11-27 at 2 55 05 PM" src="https://github.com/user-attachments/assets/db70b712-ddfa-4baa-be78-0d094ba20e59"> |
| <img width="768" alt="Screenshot 2024-11-27 at 2 56 27 PM" src="https://github.com/user-attachments/assets/ca884307-e5ff-40fe-bb4d-4fcc0e7a4f21"> |<img width="767" alt="Screenshot 2024-11-27 at 2 56 34 PM" src="https://github.com/user-attachments/assets/a9e773f6-6eec-4cc5-ac69-d48b9be5a37c"> |
| <img width="400" alt="Screenshot 2024-11-27 at 2 33 14 PM" src="https://github.com/user-attachments/assets/4134db58-c78b-4763-a0d9-be2e3e6ecc9f"> | <img width="399" alt="Screenshot 2024-11-27 at 2 33 19 PM" src="https://github.com/user-attachments/assets/54d8a825-10d5-4050-b5f1-52aba25422ce"> |
| <img width="768" alt="Screenshot 2024-11-27 at 2 59 12 PM" src="https://github.com/user-attachments/assets/c3ea0f8e-f98d-47ff-83c6-43d6388f78e9"> | <img width="769" alt="Screenshot 2024-11-27 at 2 59 07 PM" src="https://github.com/user-attachments/assets/19543ed3-6357-47e7-a61e-7eb7a8ae1765"> |

micahchiang committed:
update injected header and footer

Signed-off-by: Micah Chiang <15097156+micahchiang@users.noreply.github.com>

micahchiang commented:
@randimays - Thanks for the solid QA work. I appreciate it. I've attached more screenshots below that address the feedback you gave for the injected header and footer. Font for the `here's how you know` button has been updated and computes to Source Sans Pro like it should. 

In most other instances, I've gotten as close as possible to production, usually within 0.5px - 1px. Reason for this is I'd like to stick with design system utility classes, and TeamSites still uses a 10px root font size.

<details><summary>Official Gov Site expandable banner</summary>

![image](https://github.com/user-attachments/assets/2add9530-ae9f-427e-8219-f11db39d4216)

</details>

I only noticed the search button alignment issue on very narrow screens, < 320px, so I added a media query to adjust the margin by 1 px
<details><summary>Search bar search text and button alignment</summary>

![image](https://github.com/user-attachments/assets/4ceb3fce-0a45-459f-8d91-ee1b8cbb813c)

</details>

For the footer links, prod has a bottom margin of 16px, which leads to the gap between the links and the banner. I went with 15px, which was the closest possible value still using utility classes. Next value up would have been 20px, which looks fine but isn't as close to production currently. 
<details><summary>Spacing between footer links and PACT act banner</summary>

![image](https://github.com/user-attachments/assets/d1f967a6-7f21-46a4-8e3a-a77b2c1c9352)

</details>

randimays commented:
Thanks @micahchiang. #6 from my above comment is referring specifically to tablet/desktop spacing between the footer and the PACT Act banner. Was there an adjustment made there?

Only other concern are the missing icons on the Facility Locator search dropdown since they were not in the review instance last I checked. Can re-verify after this RI is ready (likely next week).

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
update footer container margin class

Signed-off-by: Micah Chiang <15097156+micahchiang@users.noreply.github.com>

micahchiang committed:
Merge branch 'mc-core-style-test' of github.com:department-of-veterans-affairs/vets-website into mc-core-style-test

randimays commented:
@micahchiang Hm, I am a bit confused. The Facility Locator code lives in vets-website. The icons it needs should live only in vets-website unless those icons are also used in content-build, correct? I checked the branch for [your content-build PR](https://github.com/department-of-veterans-affairs/content-build/pull/2372) and there aren't any uses of those icons.

micahchiang commented:
@randimays - hah, your guess is as good as mine :shrug: I think it's possible assets are generated from both repositories, but not 100 percent sure tbh.

micahchiang committed:
Merge branch 'main' into mc-core-style-test

randimays commented:
@micahchiang I'd advocate for not adding anything to content-build that isn't explicitly used in that repo. vets-website should be pulling everything from the s3 bucket and not from anything content-build is generating.

micahchiang commented:
@randimays - I hear you. I'm going to do a little more digging to see what I can find. It may be that review instances for vets-website aren't generating assets the same way content-build review instances are. For now though, here's the content-build review instance, where icons are properly generating for the drop down - http://6ef3cde075d3b12b0dc0d4f9af7749f5.review.vetsgov-internal/find-locations/

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
bump versions and remove formation

Signed-off-by: Micah Chiang <15097156+micahchiang@users.noreply.github.com>

micahchiang committed:
Merge branch 'mc-core-style-test' of github.com:department-of-veterans-affairs/vets-website into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
add js for footer

Signed-off-by: Micah Chiang <15097156+micahchiang@users.noreply.github.com>

micahchiang committed:
Merge branch 'mc-core-style-test' of github.com:department-of-veterans-affairs/vets-website into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang committed:
Merge branch 'main' into mc-core-style-test

micahchiang commented:
@randimays - following up here. I merged the content-build PR, and the review instance for this vets-website PR is now properly rendering the select icons in facilities - http://2b2f2e9dcf69335b1a617aa9fee24a23.review.vetsgov-internal/find-locations

randimays reviewed (approved):
LGTM

simiadebowale commented:
LGTM for VAOS/Appointments

eselkin reviewed (approved):
VAMC System/VAMC Facility/Vet Center/VBA/Facility Locator/Old FL detail pages look ok too.
